src/Utils.php

```diff
@@ -5,23 +5,58 @@
 namespace Keboola\GenericExtractor;
 
 use GuzzleHttp\Psr7\Query;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
 
 class Utils
 {
+    public static function getResource(UriInterface $uri): string
+    {
+        return $uri->getPath() . ($uri->getQuery() ? '?' . $uri->getQuery() : '');
+    }
+
     /**
-     * Merge HTTP queries, query1 values take precedence over query2 values.
+     * Merge HTTP queries, query2 values take precedence over query1 values.
      * @param string|array $query1
      * @param string|array $query2
+     * @param bool $mergeToArray
      */
-    public static function mergeQueries($query1, $query2): string
+    public static function mergeQueries($query1, $query2, bool $mergeToArray = false): string
     {
         $query1 = is_array($query1) ? $query1 : Query::parse($query1);
         $query2 = is_array($query2) ? $query2 : Query::parse($query2);
+        $mergeToArrayValues = [];
 
         foreach ($query2 as $key => $value) {
-            $query1[$key] = $value;
+            if (array_key_exists($key, $query1) && $mergeToArray) {
```

Comment:
Zatial je to tu takto zlozito, ako som pisal vyssie, je v tom chaos, upravi sa neskor.